sourcegraph/eclipse
# Secrets: move to webview login page

Previously, Eclipse used a custom native "Log in" button that was awkward to use and wasn't properly connected to the agent process. Now, we use the new `secrets: 'client-managed'` capability in the agent protocol that allows Eclipse to get/set/delete secrets. This means that the webview can just run like in VS Code and Eclipse only stores the secret.

This simplifies a lot of things, and allows us to delete all the Eclipse-specific code related to managing Sourcegraph accounts.

Requires the PR https://github.com/sourcegraph/cody/pull/5348 to be merged.
## Test plan

* Start in debug mode
* Confirm that the webview starts immediately. There is no "Log in" button anymore.
* Confirm that you can paste the access token into the webview.
* Confirm everything works as expected
* Restart Eclipse
* Confirm you're automatically signed in
* Confirm everything works as expected
* Go to settings, sign out
* Confirm you are signed out
* Restart Eclipse
* Confirm you are still signed out
